src/shims/posix/linux/foreign_items.rs

```diff
@@ -134,7 +135,12 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
                 let sys_getrandom = this.eval_libc("SYS_getrandom")?.to_machine_usize(this)?;
 
-                let sys_statx = this.eval_libc("SYS_statx")?.to_machine_usize(this)?;
+                let sys_statx = if target_os == "android" {
+                    // libc's Android target currently does not support SYS_statx -- fail gracefully
```

Comment:
Isn't this doing a syscall basically directly to the kernel?

If we need such an `if`, this is a strong indication that we should *not* share this code between android and linux, IMO.

Reply:
That's correct. This special case is required as the `libc` crate does not support a lot of constants and structures for Android even though they *do* exist, someone just has to make a PR. If you agree we should copy the Linux handlers to the Android `foreign_items` to separate them then I can just remove this condition.